src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/idam/cache/IdamCachedClientTest.java

```diff
@@ -92,21 +95,27 @@ public void should_cache_by_jurisdiction() {
         String jurisdiction2 = "cmc";
 
         given(users.getUser(jurisdiction1)).willReturn(new Credential(USERNAME, PASSWORD));
-        given(users.getUser(jurisdiction2)).willReturn(new Credential(USERNAME + 2, PASSWORD + 2));
+        given(idamApi.authenticateUser(USERNAME, PASSWORD)).willReturn(JWT);
+        given(idamApi.getUserDetails(JWT)).willReturn(USER_DETAILS);
 
-        given(idamClient.authenticateUser(USERNAME, PASSWORD)).willReturn(JWT);
-        given(idamClient.authenticateUser(USERNAME + 2, PASSWORD + 2)).willReturn(JWT2);
+        UserDetails expectedUserDetails2 =  new UserDetails("12","q@a.com","","",null);
+        given(users.getUser(jurisdiction2)).willReturn(new Credential(USERNAME + 2, PASSWORD + 2));
+        given(idamApi.authenticateUser(USERNAME + 2, PASSWORD + 2)).willReturn(JWT2);
+        given(idamApi.getUserDetails(JWT2)).willReturn(expectedUserDetails2);
 
-        String token1 =
-            idamCachedClient.getAccessToken(jurisdiction1);
-        String token2 =
-            idamCachedClient.getAccessToken(jurisdiction2);
+        CachedIdamCredential cachedIdamCredential1 =
+            idamCachedClient.getIdamCredentials(jurisdiction1);
+        CachedIdamCredential cachedIdamCredential2 =
+            idamCachedClient.getIdamCredentials(jurisdiction2);
 
-        assertThat(JWT).isEqualTo(token1);
-        assertThat(JWT2).isEqualTo(token2);
+        assertThat(cachedIdamCredential1.accessToken).isEqualTo(JWT);
+        assertThat(cachedIdamCredential1.userDetails).isEqualTo(USER_DETAILS);
```

Comment:
Worth replacing with `.isEqualTo(expectedUserDetails1`. This is not a part of this PR, though.

Reply:
changed